src/aten/sycl/Dropout.cpp

```diff
@@ -0,0 +1,488 @@
+                  scalar_t,
+                  accscalar_t,
+                  index_type,
+                  1,
+                  4,
+                  mask_t>(
+                  self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+              sycl_kernel_submit(
+                  num_groups * group_size,
+                  group_size,
+                  getCurrentSYCLQueue(),
+                  caller);
+            } break;
+            case 2: {
+              auto caller = FusedDropoutVecFunctor<
+                  scalar_t,
+                  accscalar_t,
+                  index_type,
+                  1,
+                  2,
+                  mask_t>(
+                  self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+              sycl_kernel_submit(
+                  num_groups * group_size,
+                  group_size,
+                  getCurrentSYCLQueue(),
+                  caller);
+            } break;
+          }
+        } else {
+          switch (self_info.dims) {
+            case 1: {
+              auto caller = FusedDropoutUnrollFunctor<
+                  scalar_t,
+                  accscalar_t,
+                  index_type,
+                  1,
+                  1,
+                  mask_t>(
+                  self_info, ret_info, mask_info, nelem, pa, rng_engine_inputs);
+              sycl_kernel_submit(
+                  num_groups * group_size,
+                  group_size,
+                  getCurrentSYCLQueue(),
+                  caller);
+            } break;
+            default:
+              if (!self.is_contiguous() && ret.is_contiguous() &&
+                  mask.is_contiguous()) {
+                auto caller = FusedDropoutUnrollFunctor<
+                    scalar_t,
+                    accscalar_t,
+                    index_type,
+                    -1,
+                    1,
+                    mask_t>(
+                    self_info,
+                    ret_info,
+                    mask_info,
+                    nelem,
+                    pa,
+                    rng_engine_inputs);
+                sycl_kernel_submit(
+                    num_groups * group_size,
+                    group_size,
+                    getCurrentSYCLQueue(),
+                    caller);
+              } else {
+                auto caller = FusedDropoutUnrollFunctor<
+                    scalar_t,
+                    accscalar_t,
+                    index_type,
+                    -1,
+                    -1,
+                    mask_t>(
+                    self_info,
+                    ret_info,
+                    mask_info,
+                    nelem,
+                    pa,
+                    rng_engine_inputs);
+                sycl_kernel_submit(
+                    num_groups * group_size,
+                    group_size,
+                    getCurrentSYCLQueue(),
+                    caller);
+              }
+          }
```

Comment:
Please fine the logic.

Reply:
The logic is kind of complex. Why could we not use C++ template Traits or other template tech to make it neat?

Reply:
Since switch is another kind of if-else. It means the code block contains three nested level if-else.